### PR TITLE
googleapis: conan v2 support + add test_v1_package

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@jgsogo `helpers.py` doesn't seem to be robust to a layout where source code is at the same level than CMakeLists, could you help here?

**Comment by jgsogo** [comment]
> @jgsogo `helpers.py` doesn't seem to be robust to a layout where source code isn't at the same level than CMakeLists, could you help here?

Probably this one is enough: https://github.com/SpaceIm/conan-center-index/pull/14

**Comment by jgsogo** [comment]
Conflicts with `master`, just some bumped dependencies

File: recipes/googleapis/all/conanfile.py [hunk 1]
```diff
@@ -119,40 +132,47 @@ def deactivate_library(key):
         #  - Inconvenient macro names from usr/include/sys/syslimits.h in some macOS SDKs: GID_MAX
         #    Patched here: https://github.com/protocolbuffers/protobuf/commit/f138d5de2535eb7dd7c8d0ad5eb16d128ab221fd
         #    as of 3.21.4 issue still exist
-        if tools.Version(self.deps_cpp_info["protobuf"].version) <= "3.21.5" and self.settings.os == "Macos":
+        if Version(self.deps_cpp_info["protobuf"].version) <= "3.21.5" and self.settings.os == "Macos":
```
Comment (SpaceIm) [inline recipes/googleapis/all/conanfile.py:135]: should be replace by `self.dependencies["protobuf"].ref.version` I guess, but according to https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface, it should be used in `validate()` and `generate()` only, and here we are in `build()`.

**Comment by SpaceIm** [comment]
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(382,5): error MSB3491: Could not write lines to file "google_cloud_beyondcorp_clientconnectorservices_v1_clientconnectorservices_proto.dir\Debug\google_c.F8B155F2.tlog\google_cloud_beyondcorp_clientconnectorservices_v1_clientconnectorservices_proto.lastbuildstate". Path: google_cloud_beyondcorp_clientconnectorservices_v1_clientconnectorservices_proto.dir\Debug\google_c.F8B155F2.tlog\google_cloud_beyondcorp_clientconnectorservices_v1_clientconnectorservices_proto.lastbuildstate exceeds the OS max path limit. The fully qualified file name must be less than 260 characters. [C:\J\w\prod\BuildSingleReference@4\s\34573a\1\build\google_cloud_beyondcorp_clientconnectorservices_v1_clientconnectorservices_proto.vcxproj]
```

:sweat:

**Comment by jgsogo** [comment]
What do the new layouts introduce? Maybe it is just the `Debug` extra item in the path?  🤕

**Comment by SpaceIm** [comment]
> What do the new layouts introduce? Maybe it is just the `Debug` extra item in the path? 🤕

I'll try to redefine build_folder in layout to reduce path length.

**Pull request closed without merging**